src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java

```diff
@@ -169,6 +170,31 @@ public FeedbackResponseCommentAttributes updateFeedbackResponseComment(
             throw new InvalidParametersException(newAttributes.getInvalidityInfo());
         }
 
+        // update only if change
+        boolean hasSameAttributes =
+                this.<String>hasSameValue(frc.getFeedbackResponseId(), newAttributes.getFeedbackResponseId());
+        hasSameAttributes = hasSameAttributes
+                && this.<String>hasSameValue(frc.getCommentText(), newAttributes.getCommentText());
+        hasSameAttributes = hasSameAttributes
+                && this.<List<FeedbackParticipantType>>hasSameValue(
+                        frc.getShowCommentTo(), newAttributes.getShowCommentTo());
+        hasSameAttributes = hasSameAttributes
+                && this.<List<FeedbackParticipantType>>hasSameValue(
+                        frc.getShowGiverNameTo(), newAttributes.getShowGiverNameTo());
+        hasSameAttributes = hasSameAttributes
+                && this.<String>hasSameValue(frc.getLastEditorEmail(), newAttributes.getLastEditorEmail());
+        hasSameAttributes = hasSameAttributes
+                && this.<Instant>hasSameValue(frc.getLastEditedAt(), newAttributes.getLastEditedAt());
+        hasSameAttributes = hasSameAttributes
+                && this.<String>hasSameValue(frc.getGiverSection(), newAttributes.getGiverSection());
+        hasSameAttributes = hasSameAttributes
+                && this.<String>hasSameValue(frc.getReceiverSection(), newAttributes.getReceiverSection());
```

Comment:
Why not just say 
```java
if (hasSameValue(A) && hasSameValue(B) ... ) {
  // apply optimize policy
}
```
 instead of using a variable and do so many operations

The `if` statement can perform shortcut for boolean operation, and will be further optimized

Or, if you think the long `if` statement is not clear for other developers, how about abstract this hunk for a method:
```java
private boolean isNewAttributeSameAsOldOne(newAttr, oldAttr) {
//...
}
```

Reply:
> The if statement can perform shortcut for boolean operation, and will be further optimized

Shortcut evaluation is also enabled for variable assignment. You don't need an long if to enable it. 

> how about abstract this hunk for a method:

I can try to abstract the chunk to method.